crates/core/src/metadata_store/providers/objstore/object_store_version_repository.rs

```diff
@@ -56,8 +62,15 @@ impl ObjectStoreVersionRepository {
         let store = object_store::memory::InMemory::new();
         Self {
             object_store: Box::new(store),
+            prefix: Default::default(),
         }
     }
+
+    /// Convert a metadata store key into an object store path.
+    #[inline]
+    fn path(&self, key: &ByteString) -> Path {
+        self.prefix.child(key.to_string())
```

Comment:
`<ByteString as AsRef<str>>::as_ref(key)` saves a string allocation

Reply:
Thank you for the tip! :-)